File: app/src/main/java/com/macv/wheeldemo/MainActivity.java
package com.macv.wheeldemo;

import android.graphics.BitmapFactory;
import android.graphics.Color;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.macv.mwheel.MSpinWheel;
import com.macv.mwheel.WheelListener;
import com.macv.mwheel.WheelItem;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class MainActivity extends AppCompatActivity {

    private MSpinWheel lw;
    List<WheelItem> wheelItems ;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        generateWheelItems();

        lw = findViewById(R.id.lwv);
        lw.addWheelItems(wheelItems);


        lw.setWheelListener(new WheelListener() {
            @Override
            public void onOpen() {

            }

            @Override
            public void onClose() {

            }

            @Override
            public void onReward(int pos) {
                Toast.makeText(MainActivity.this, "Target Reached "+wheelItems.get(pos).text, Toast.LENGTH_LONG).show();
            }

            @Override
            public void onStart() {
                Toast.makeText(MainActivity.this, "Start " , Toast.LENGTH_LONG).show();
            }

            @Override
            public void onError() {

            }
        });

        Button start = findViewById(R.id.start);
        start.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                lw.rotateWheelTo(new Random().nextInt(wheelItems.size()));
            }
        });

    }

    private void generateWheelItems() {
        wheelItems = new ArrayList<>();
        wheelItems.add(new WheelItem(Color.parseColor("#141A1E"), BitmapFactory.decodeResource(getResources(),R.drawable.chat) , "100 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#040D14"), BitmapFactory.decodeResource(getResources(),R.drawable.coupon) , "0 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#141A1E"), BitmapFactory.decodeResource(getResources(),R.drawable.ice_cream), "30 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#040D14"), BitmapFactory.decodeResource(getResources(),R.drawable.lemonade), "6000 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#141A1E"), BitmapFactory.decodeResource(getResources(),R.drawable.orange), "9 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#040D14"), BitmapFactory.decodeResource(getResources(),R.drawable.shop), "20 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#141A1E"), BitmapFactory.decodeResource(getResources(),R.drawable.orange), "9 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#040D14"), BitmapFactory.decodeResource(getResources(),R.drawable.shop), "20 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#141A1E"), BitmapFactory.decodeResource(getResources(),R.drawable.orange), "9 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#040D14"), BitmapFactory.decodeResource(getResources(),R.drawable.shop), "20 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#141A1E"), BitmapFactory.decodeResource(getResources(),R.drawable.orange), "9 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#040D14"), BitmapFactory.decodeResource(getResources(),R.drawable.shop), "20 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#141A1E"), BitmapFactory.decodeResource(getResources(),R.drawable.orange), "9 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#040D14"), BitmapFactory.decodeResource(getResources(),R.drawable.shop), "20 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#141A1E"), BitmapFactory.decodeResource(getResources(),R.drawable.orange), "9 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#040D14"), BitmapFactory.decodeResource(getResources(),R.drawable.shop), "20 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#141A1E"), BitmapFactory.decodeResource(getResources(),R.drawable.orange), "9 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#040D14"), BitmapFactory.decodeResource(getResources(),R.drawable.shop), "20 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#141A1E"), BitmapFactory.decodeResource(getResources(),R.drawable.orange), "9 $"));
        wheelItems.add(new WheelItem(Color.parseColor("#040D14"), BitmapFactory.decodeResource(getResources(),R.drawable.shop), "20 $"));
    }
}

File: app/src/main/java/com/macv/wheeldemo/MainActivity.kt
package com.macv.wheeldemo

import android.graphics.Color
import android.os.Bundle
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.macv.mwheel.MSpinWheel
import com.macv.mwheel.WheelItem
import com.macv.mwheel.WheelListener
import java.util.*

class MainActivity : AppCompatActivity() {
    private lateinit var lw: MSpinWheel
    private lateinit var wheelItems: ArrayList<WheelItem>
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        generateWheelItems()
        lw = findViewById(R.id.lwv)
        lw.addWheelItems(wheelItems)
        lw.setWheelListener(object : WheelListener {
            override fun onOpen() {
                Toast.makeText(this@MainActivity, "Open ", Toast.LENGTH_LONG).show()
            }
            override fun onClose() {

            }
            override fun onReward(pos: Int) {
                Toast.makeText(this@MainActivity, "Target Reached " + wheelItems[pos].displayText, Toast.LENGTH_LONG).show()
            }

            override fun onStart() {
                Toast.makeText(this@MainActivity, "Start ", Toast.LENGTH_LONG).show()
            }

            override fun onError(s: String) {
                Toast.makeText(this@MainActivity, s, Toast.LENGTH_LONG).show()
            }
        })
        val start = findViewById<Button>(R.id.start)
        start.setOnClickListener { lw.rotateWheelTo(Random().nextInt(wheelItems.size)) }
    }

    private fun generateWheelItems() {
        wheelItems = ArrayList()
        wheelItems.add(WheelItem(Color.parseColor("#141A1E"), "1 $", 1, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#040D14"), "2 $", 2, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#141A1E"), "3 $", 3, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#040D14"), "4 $", 4, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#141A1E"), "5 $", 5, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#040D14"), "6 $", 6, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#141A1E"), "7 $", 7, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#040D14"), "8 $", 8, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#141A1E"), "9 $", 9, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#040D14"), "10 $", 10, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#141A1E"), "11 $", 11, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#040D14"), "12 $", 12, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#141A1E"), "13 $", 13, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#040D14"), "14 $", 14, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#141A1E"), "15 $", 15, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#040D14"), "16 $", 16, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#141A1E"), "17 $", 17, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#040D14"), "18 $", 18, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#141A1E"), "19 $", 19, "USD"))
        wheelItems.add(WheelItem(Color.parseColor("#040D14"), "20 $", 20, "USD"))
    }
}